pkg/apis/build/v1alpha2/build_validation.go

```diff
@@ -0,0 +1,98 @@
+package v1alpha2
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/kmp"
+
+	"github.com/pivotal/kpack/pkg/apis/validate"
+)
+
+func (b *Build) SetDefaults(ctx context.Context) {
+	if b.Spec.ServiceAccount == "" {
+		b.Spec.ServiceAccount = "default"
+	}
+
+	for i, s := range b.Spec.Services {
+		if s.Kind == "Secret" && s.APIVersion == "" {
+			b.Spec.Services[i].APIVersion = "v1"
```

Comment:
is there precedent for defaulting the APIVersion in an object reference?

Reply:
This was just me speculating that this might help usability